loccallibrary/src/controllers/author.controller.ts

```diff
@@ -52,26 +60,118 @@ export const authorUpdatePost = asyncHandler(
   }
 );
 
-export const authorDeletePost = asyncHandler(
+export const authorDeleteGet = asyncHandler(
   async (req: Request, res: Response) => {
-    res.send(`NOT IMPLEMENTED: Author delete: ${req.params.id}`);
+    const id = parseInt(req.params.id);
+    const { t } = req;
+    if (isNaN(id)) {
+      return res.render("./error", {
+        error: {
+          status: 404,
+          message: t("author.error_message_invalid"),
+        },
+      });
+    }
+    const author = await findById(id);
+    if (author === null) {
+      res.redirect("/authors/author");
+    }
+    const allBooksByAuthor = author?.books;
+    res.render("authors/delete", {
+      title: "Delete Author",
+      author: author,
+      books: allBooksByAuthor,
+    });
   }
 );
 
-export const authorCreatePost = asyncHandler(
+export const authorDeletePost = asyncHandler(
   async (req: Request, res: Response) => {
-    res.send("NOT IMPLEMENTED: Author create");
-  }
-);
+    const id = parseInt(req.params.id);
+    if (isNaN(id)) {
+      return res.render("./error", {
+        error: {
+          status: 404,
+          message: t("author.error_message_invalid"),
+        },
+      });
+    }
 
-export const authorCreateGet = asyncHandler(
-  async (req: Request, res: Response) => {
-    res.send("NOT IMPLEMENTED: Author create");
+    const author = await findById(id);
+    if (author === null) {
+      res.redirect("/authors");
+    }
+
+    const allBooksByAuthor = author?.books;
+
+    if (allBooksByAuthor.length > 0) {
+      res.render("authors/delete", {
+        title: "Delete Author",
+        author: author,
+        authorBooks: allBooksByAuthor,
+      });
+      return;
+    } else {
+      await deleteById(id);
+      res.redirect("/authors/list");
+    }
+
+    const authors = await listAuthors();
+    return res.render("authors/author", { authors });
   }
 );
 
-export const authorDeleteGet = asyncHandler(
+export const authorCreatePost = [
+  body("firstName", "first name must not be empty")
+    .trim()
+    .isLength({ min: 1 })
+    .escape(),
+  body("familyName", "family name must not be empty")
+    .trim()
+    .isLength({ min: 1 })
+    .escape(),
+  body("dateOfBirth", "Date of Birth must not be empty")
+    .optional({ checkFalsy: false })
+    .isDate(),
+  body("dateOfDeath", "Date of Death must not be empty")
+    .optional({ checkFalsy: false })
+    .isDate(),
```

Comment:
phần xử lý validate này e move ra function not export để tái sử dụng nha
Sửa tương tự trong các file controller bên dưới nhé